### DISCUSSION TITLE
Language management

### DISCUSSION THREAD

**Discussion by ClemTB** [reorproject/reor#187] [Ideas]
I think Reor is lacking of language management, so it can be nice :
* to detect each note language and tag the note with it so the LLM knows better how to handle it
* To provide a user language so the LLM responds in that language
* Of course translate the UI ^^

Hope this helps

**Comment by samlhuillier** [comment]
Thank you for suggesting this and apologies for the delay.

In the coming weeks we will be working on a feature to allow users to customise prompts completely (both system and user). This should make the models respond more naturally in the language of choice...

On translating the UI, this is not something we really have bandwidth for right now...

**Reply by ClemTB** [reply]
It's ok, I did not even hoped for an answer.
Implement custom prompt is a good thing for sure. 
Also, I think the suggestion about having a specific language for each note is a nice step: 
1) English is not my mother tongue, so I write notes in french, but most of the notes I "import" are in english
1) Reor can manage notes by language, so you can ask to search/treat only japanese notes
2) Prepare the llm before every note with proper language will probably make better comprehension of it

In the same way, a tag mechanism can be a great feature, and the language management a part of it. I can think of notes that are tagged with languages, topics, ... 
Also, tags could be added by the llm itself if requested (with promt like "summarise this text with at most 5 words" and "Tell me in what language is this text written")

**Reply by BradKML** [reply]
TBH I would need the same tagging system but for tracking different agents and what philosophical "stance" they take when it comes to writing, or if they have been "digested" by an AI summarizer vs just "parsed" from file
